stake-pool/js/test/stakepool-program.test.ts

```diff
@@ -0,0 +1,255 @@
+import {
+  PublicKey,
+  Connection,
+  Keypair,
+  clusterApiUrl,
+  SystemProgram, AccountInfo, LAMPORTS_PER_SOL
+} from '@solana/web3.js';
+import {
+  STAKE_POOL_INSTRUCTION_LAYOUTS,
+  DepositSolParams,
+  StakePoolProgram,
+} from '../src/stakepool-program';
+import { decodeData } from '../src/copied-from-solana-web3/instruction';
+import { depositSol, withdrawSol, withdrawStake } from "../src";
+import { STAKE_POOL_LAYOUT } from "../src/layouts";
+import { mockTokenAccount, mockValidatorList, stakePoolMock } from "./mocks";
+
+describe('StakePoolProgram', () => {
+
+  const connection = new Connection(
```

Comment:
Does this ever actually call out to devnet for anything? it looks like you've mocked just about every part of it, which is awesome!